dotnet/src/Connectors/VectorData.Abstractions/VectorStorage/VectorStore.cs

```diff
@@ -31,10 +32,20 @@ public abstract class VectorStore : IDisposable
     /// <seealso cref="VectorStoreKeyAttribute"/>
     /// <seealso cref="VectorStoreDataAttribute"/>
     /// <seealso cref="VectorStoreVectorAttribute"/>
+    [RequiresDynamicCode("This API is not compatible with NativeAOT. For dynamic mapping via Dictionary<string, object?>, use GetCollectionDynamic() instead.")]
+    [RequiresUnreferencedCode("This API is not compatible with trimming. For dynamic mapping via Dictionary<string, object?>, use GetCollectionDynamic() instead.")]
     public abstract VectorStoreCollection<TKey, TRecord> GetCollection<TKey, TRecord>(string name, VectorStoreRecordDefinition? vectorStoreRecordDefinition = null)
         where TKey : notnull
         where TRecord : class;
 
+    /// <summary>
+    /// Gets a collection from the vector store, using dynamic mapping; the record type is represented as a <see cref="Dictionary{TKey, TValue}"/>.
+    /// </summary>
+    /// <param name="name">The name of the collection.</param>
+    /// <param name="vectorStoreRecordDefinition">The schema of the record type.</param>
+    /// <returns>A new <see cref="VectorStoreCollection{TKey, TRecord}"/> instance for managing the records in the collection.</returns>
+    public abstract VectorStoreCollection<object, Dictionary<string, object?>> GetDynamicCollection(string name, VectorStoreRecordDefinition vectorStoreRecordDefinition);
```

Comment:
It's a pity that we need a new method and bunch of derived types, but I trust your experience and choosing the best available solution.

Reply:
I'm definitely open to discussing if you have other ideas! But to my understanding the basic requirements here are:
1. Ensure that users of an unsafe codepath (i.e. POCO mapping without source generation) get warnings.
2. Users of safe codepaths (dynamic mapping, POCO mapping with source generation) do not get warnings.

AFAIK that means having different entry points for the above cases (a total of 3, for now only 2 since we don't yet have source generation). I know it's not great to add more public API surface for this, but at least it stays out of the way for people who don't care about it...

Let me know your thoughts, or I'm happy to jump onto a call to discuss further.